app/controllers/groups_controller.rb

```diff
@@ -286,22 +288,47 @@ def leave
     if cud
       leaver = @assessment.aud_for(cud.id)
       booter = @assessment.aud_for(@cud.id)
-      if @group.is_member(booter) && leaver.group_id == @group.id &&
-         leaver.membership_status != AssessmentUserDatum::CONFIRMED
-        leaver.group = nil
-        leaver.membership_status = AssessmentUserDatum::UNCONFIRMED
-        leaver.save!
+
+      # Can remove if the booter is a member of the group and the leaver is not confirmed
+      can_remove = @group.is_member(booter) &&
+                   leaver.membership_status != AssessmentUserDatum::CONFIRMED
+      # ... or if the booter is an instructor
+      can_remove ||= @cud.instructor
+      # ... but only if the leaver is in the group
+      can_remove &&= leaver.group_id == @group.id
+
+      if can_remove
+        leaver.leave_group
+        if @group.is_effectively_empty
+          @group.destroy!
+          flash[:success] = "Successfully removed #{cud.user.full_name} from group and disbanded " \
+                            "group as all remaining members (if any) require group confirmation."
+        else
+          flash[:success] = "Successfully removed #{cud.user.full_name} from group."
+        end
+
       end
     else
       leaver = @assessment.aud_for(@cud.id)
-      redirect_to([@course, @assessment, @group]) && return unless leaver.group_id == @group.id
+      flash.delete(:error) # clear the error from get_member_cud
+
+      unless leaver.group_id == @group.id
+        flash[:error] = "You are not a member of this group."
+        redirect_to(course_assessment_group_path(@course, @assessment, @group)) && return
+      end
 
-      leaver.group = nil
-      leaver.membership_status = AssessmentUserDatum::UNCONFIRMED
       ActiveRecord::Base.transaction do
-        leaver.save!
-        @group.destroy! if @group.assessment_user_data.empty?
+        leaver.leave_group
+        if @group.is_effectively_empty
+          @group.destroy!
+          flash[:success] =
+            "Successfully left and disbanded group as all remaining members (if any) "\
```

Comment:
I don't think this works, when I create a group, invite someone who has not accepted and leave the group. The group is not disbanded, it still exists.

Reply:
This works as expected. The person you invited can still accept your invite and join the group.